src/github.py

```diff
@@ -8,6 +8,30 @@ def __init__(self, group: str, repo: str, auth_token: str = None) -> None:
         self.repo: str = repo
         self.auth_token: str = auth_token
 
+    def get_test_result(test, logfiles, col):
```

Comment:
I think this function should not be in GitHubRepo, because it has nothing to do with a GitHub repo.

Reply:
This function does two things, check if a test exists, and report their collective status. Consider splitting them.

Reply:
I moved get_test_result to ResultList. It now checks only the color of for the test. If the test doesn't exist, the color is None. Thus, test_exist became redundant.